mutation/types.ts

```diff
@@ -56,7 +56,7 @@ interface TriggerWithArgs<
   ExtraArg = never
 > {
   <SWRData = Data>(
-    extraArgument: ExtraArg,
+    extraArgument?: ExtraArg,
```

Comment:
I don't think this is the right fix. And it breaks the type validation for required `args`

What we expect here are 

For required `arg`

```tsx
const { trigger, reset, data, error } = useSWRMutation(
    '/api/user',
    (_, opts: { arg: number }) => String(opts.arg)
)
// should throw error, but your change breaks this.
trigger()
// pass type validation
trigger(1)
```

For optional `arg`

```tsx
const { trigger, reset, data, error } = useSWRMutation(
    '/api/user',
    (_, opts: { arg?: number }) => String(opts.arg)
)
// pass type validation
trigger()
// pass type validation
trigger(1)
```